File: src/strategy/pedro/persona/Persona.java
package strategy.pedro.persona;

import strategy.pedro.bebida.Bebida;
import strategy.pedro.bebida.NingunaBebida;

public abstract class Persona {
    protected String nombre;
    private Bebida bebida = new NingunaBebida();

    public void muestra() {
        System.out.println("Hola soy un " + nombre);
    }

    public void bebe() {
        System.out.println("Bebo " + bebida.get());
    }

    public void setBebida(Bebida bebida) {
        this.bebida = bebida;
    }
}

File: manuel/strategy/Persona.php
<?php
    abstract class Persona{
        protected $bebida;
        protected $edad;
        protected $nombre;

        public function beber(){
            if ($this->edad < $this->bebida->getEdadBase()){
                echo "No puedo beber ".strtolower($this->bebida->getNombre());
            }else{
                echo "Hola soy una persona de ".$this->edad." años y estoy 
                    bebiendo ".strtolower($this->bebida->getNombre());
            }
        }

        public function setBebida(Bebida $bebida){
            $this->bebida = $bebida;
        }
    }
?>